src/libs/actions/IOU.ts

```diff
@@ -2314,6 +2327,23 @@ function calculateDiffAmount(iouReport: OnyxEntry<OnyxTypes.Report>, updatedTran
     return 0;
 }
 
+function calculateAmountForUpdatedWaypoint(transaction: OnyxEntry<OnyxTypes.Transaction>, transactionChanges: TransactionChanges, policy: OnyxEntry<OnyxTypes.Policy>) {
+    let updatedAmount: number = CONST.IOU.DEFAULT_AMOUNT;
+    let updatedMerchant = Localize.translateLocal('iou.fieldPending');
+    if (transactionChanges?.routes) {
+        const customUnitRateID = TransactionUtils.getRateID(transaction) ?? '';
+        const mileageRates = DistanceRequestUtils.getMileageRates(policy);
+        const mileageRate = mileageRates?.[customUnitRateID];
```

Comment:
We should use this logic which properly handles the P2P rate case.

https://github.com/Expensify/App/blob/b12858ea5078397bca91373557d6c102ec0e9336/src/components/MoneyRequestConfirmationList.tsx#L259-L261

Reply:
I updated.